agent/config/flags.go

```diff
@@ -61,7 +61,7 @@ func AddFlags(fs *flag.FlagSet, f *Flags) {
 	add(&f.Config.BootstrapExpect, "bootstrap-expect", "Sets server to expect bootstrap mode.")
 	add(&f.Config.ClientAddr, "client", "Sets the address to bind for client access. This includes RPC, DNS, HTTP, HTTPS and gRPC (if configured).")
 	add(&f.ConfigFiles, "config-dir", "Path to a directory to read configuration files from. This will read every file ending in '.json' as configuration in this directory in alphabetical order. Can be specified multiple times.")
-	add(&f.ConfigFiles, "config-file", "Path to a JSON file to read configuration from. Can be specified multiple times.")
+	add(&f.ConfigFiles, "config-file", "Path to a file with JSON or HCL extension to read configuration from. Can be specified multiple times.")
```

Comment:
```suggestion
	add(&f.ConfigFiles, "config-file", "Path to a file in JSON or HCL format to read configuration from. Can be specified multiple times.")
```

That seems to be more consistent with the description of `config-format`.

Reply:
I see what you mean. However, I think if we say the file is in JSON or HCL format, it still leaves the issue from #4506 open where: Someone has a JSON formatted file with no extension and no config-format and they expect it to be read as JSON. 

What do you think of this edit?

>Path to a file in JSON or HCL format to read configuration from, whose extension matches the file format. Can be specified multiple times.

This way the extension requirement is made explicit.

Reply:
or actually this is easier to read:

> Path to a file in JSON or HCL format with a matching file extension. Can be specified multiple times.

Reply:
👍